### PR TITLE
Fix NPE for s3 proxy v2

### PR DESCRIPTION
**Pull request by humengyu2012**
### What changes are proposed in this pull request?

Fix NPE for S3 Porxy.

### Why are the changes needed?

When we enable s3 v2 and authentication, the NullPointException will occur:
```
alluxio.s3.rest.authentication.enabled=true
alluxio.proxy.s3.v2.version.enabled=true
```
like: 
```
2023-03-14 22:54:51,738 ERROR S3Handler - Exception during create s3handler:alluxio.proxy.s3.S3Exception
	at alluxio.proxy.s3.signature.AwsSignatureProcessor.getAuthInfo(AwsSignatureProcessor.java:138)
	at alluxio.proxy.s3.S3RestUtils.getUserFromSignature(S3RestUtils.java:576)
	at alluxio.proxy.s3.S3RestUtils.getUser(S3RestUtils.java:562)
	at alluxio.proxy.s3.S3Handler.doAuthentication(S3Handler.java:458)
	at alluxio.proxy.s3.S3Handler.init(S3Handler.java:230)
	at alluxio.proxy.s3.S3Handler.createHandler(S3Handler.java:147)
	at alluxio.proxy.s3.S3RequestServlet.service(S3RequestServlet.java:73)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799)
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1631)
	at alluxio.web.CORSFilter.doFilter(CORSFilter.java:54)
	at alluxio.web.HttpFilter.doFilter(HttpFilter.java:48)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:548)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:600)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1440)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:59)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
	at alluxio.proxy.s3.signature.StringToSignProducer.createSignatureBase(StringToSignProducer.java:78)
	at alluxio.proxy.s3.signature.AwsSignatureProcessor.getAuthInfo(AwsSignatureProcessor.java:118)
	... 42 more
```

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs
  2. addition or removal of property keys
  3. webui

### CODE REVIEW

File: core/server/proxy/src/main/java/alluxio/proxy/s3/signature/StringToSignProducer.java [hunk 1]
```diff
@@ -83,6 +88,27 @@ public static String createSignatureBase(
...
+   * @return signature string
+   * @throws Exception
+   */
+  public static String createSignatureBase(
+      SignatureInfo signatureInfo,
+      HttpServletRequest request
+  ) throws Exception {
+    URI uri = new URI(request.getRequestURL().toString());
+    return createSignatureBase(signatureInfo,
+        uri.getScheme(),
```
Comment (lucyge2022) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/signature/StringToSignProducer.java:105]:
```suggestion
        request.getScheme()
```

File: core/server/proxy/src/main/java/alluxio/proxy/s3/signature/StringToSignProducer.java [hunk 2]
```diff
@@ -83,6 +88,27 @@ public static String createSignatureBase(
...
+        request.getMethod(),
+        uri.getPath(),
```
Comment (lucyge2022) [inline core/server/proxy/src/main/java/alluxio/proxy/s3/signature/StringToSignProducer.java:107]:
```suggestion
        request.getRequestURI()
```

**Review by lucyge2022 — APPROVED** [review]
LGTM other than a few small comment, thanks for catching the bug.

**Comment by lucyge2022** [comment]
LGTM.  Unrelated to this PR, but keep here for record:   
  in current integration test S3ClientRestApiTest we have property 'alluxio.s3.rest.authentication.enabled'  disabled to avoid preparing auth header for testing, we should make this 'alluxio.s3.rest.authentication.enabled' enabled by default, and have the other auth-bypassed route disabled or enabled by a flag. FYI @Jackson-Wang-7

**Review by lucyge2022 — APPROVED** [review]

**Comment by HelloHorizon** [comment]
alluxio-bot, merge this please

**Pull request merged**